src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/factory/generator/flow/SingleTableIngressRuleGenerator.java

```diff
@@ -0,0 +1,118 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.factory.generator.flow;
+
+import org.openkilda.Command;
+import org.openkilda.CommandType;
+import org.openkilda.Constants;
+import org.openkilda.Field;
+import org.openkilda.FlowCommand;
+import org.openkilda.FlowCommand.FlowCommandBuilder;
+import org.openkilda.FlowFlag;
+import org.openkilda.Instructions;
+import org.openkilda.MeterCommand;
+import org.openkilda.action.Action;
+import org.openkilda.action.PopVlanAction;
+import org.openkilda.action.PushVlanAction;
+import org.openkilda.adapter.FlowSideAdapter;
+import org.openkilda.match.FieldExactMatch;
+import org.openkilda.match.Match;
+import org.openkilda.model.FlowEncapsulationType;
+import org.openkilda.model.FlowEndpoint;
+import org.openkilda.model.MeterId;
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchFeature;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import lombok.experimental.SuperBuilder;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+@SuperBuilder
+public class SingleTableIngressRuleGenerator extends IngressRuleGenerator {
+
+    @Override
+    public List<Command> generateCommands(Switch sw) {
+        List<Command> result = new ArrayList<>();
+        Command meterCommand = buildMeter(flowPath.getMeterId());
+        result.add(meterCommand);
+
+        FlowSideAdapter flowSide = FlowSideAdapter.makeIngressAdapter(flowPath);
+        FlowCommand command = buildFlowCommand(sw, flowSide);
+        result.add(command);
+        if (command != null) {
+            if (meterCommand != null) {
+                command.getDependsOn().add(meterCommand.getUuid());
+            }
+        }
+
+        return result;
+    }
+
+    private Command buildMeter(MeterId meterId) {
+        // todo fix meter
+        return MeterCommand.builder().build();
+    }
+
+    private FlowCommand buildFlowCommand(Switch sw, FlowSideAdapter flowSide) {
+        FlowEndpoint endpoint = flowSide.getEndpoint();
+        Collection<Match> match = Lists.newArrayList(
+                FieldExactMatch.builder().field(Field.IN_PORT).value(endpoint.getPortNumber()).build());
+        if (isDefaultFlow(flowSide)) {
+            match.add(FieldExactMatch.builder().field(Field.VLAN_VID).value(endpoint.getOuterVlanId()).build());
+        }
+        List<Action> actions = new ArrayList<>();
+        if (!isDefaultFlow(flowSide)) {
+            actions.add(new PopVlanAction());
```

Comment:
Why? How do you know that we do not need vlan for transit encasulation?

Reply:
Added todo to fix it.